torch.Package extend PyTorchStreamWriter to track written records

lobstergrindset opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #52176 torch.Package zipfile debugging printer
* **#52218 torch.Package extend PyTorchStreamWriter to track written records**

Differential Revision: [D26429794](https://our.internmc.facebook.com/intern/diff/D26429794)

Commit:
torch.Package extend PyTorchStreamWriter to track written records

[ghstack-poisoned]

Commit:
Update on "torch.Package extend PyTorchStreamWriter to track written records"




[ghstack-poisoned]

suo commented on caffe2/serialize/inline_container.cc:
```diff
@@ -356,6 +360,7 @@ void PyTorchStreamWriter::writeRecord(
       nullptr,
       0);
   valid("writing file ", name.c_str());
+  files_written.push_back(std::move(name));
```
nit: name is a const ref, moving it unnecessary as it does not avoid the copy.

suo reviewed (approved):
cool!

Commit:
Update on "torch.Package extend PyTorchStreamWriter to track written records"


Differential Revision: [D26429794](https://our.internmc.facebook.com/intern/diff/D26429794)

[ghstack-poisoned]

Commit:
Update on "torch.Package extend PyTorchStreamWriter to track written records"


Differential Revision: [D26429794](https://our.internmc.facebook.com/intern/diff/D26429794)

[ghstack-poisoned]

Commit:
Update on "torch.Package extend PyTorchStreamWriter to track written records"


Differential Revision: [D26429794](https://our.internmc.facebook.com/intern/diff/D26429794)

[ghstack-poisoned]
